server/src/main/java/com/cloud/api/query/dao/HostJoinDaoImpl.java

```diff
@@ -274,12 +269,19 @@ public HostResponse newHostResponse(HostJoinVO host, EnumSet<HostDetails> detail
     @Override
     public HostResponse setHostResponse(HostResponse response, HostJoinVO host) {
         String tag = host.getTag();
-        if (tag != null) {
-            if (response.getHostTags() != null && response.getHostTags().length() > 0) {
+        if (StringUtils.isNotEmpty(tag)) {
+            if (StringUtils.isNotEmpty(response.getHostTags())) {
                 response.setHostTags(response.getHostTags() + "," + tag);
             } else {
                 response.setHostTags(tag);
             }
+
+            if (!Boolean.TRUE.equals(response.getHaHost())) {
```

Comment:
Suggestion:
Some ways to improve this validation:

- As it does not have an `else` operator, we could just validate with `!responsegetHaHost()`;
- We could use `commons.lang3.BooleanUtils`,  `BooleanUtils.isFalse(response.getHaHost())`;
- As it is validating if it is `false`, we could use `Boolean.FALSE.equals(response.getHaHost())`;
- Do we use the third value of a Boolean (`null`) in `haHost`, somewhere?  If not, we could do a validation on `getHaHost()`;

Reply:
@shwstppr could you take a look at this?

Reply:
my choice would be
```suggestion
            if (Boolean.FALSE.equals(response.getHaHost())) {
```
@shwstppr can you answer @GutoVeronezi ?

Reply:
@DaanHoogland I would prefer code as it is

Reply:
@shwstppr 
I do not see any technical reason to keep a `negation of a TRUE` when we could simply use `FALSE`, besides that it goes against KISS principle; but it is my opinion against yours, in your PR, so I have nothing more to say.